aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -209,6 +209,12 @@
    */
   public static final String CLIENT_ASSUME_ROLE_REGION = "client.assume-role.region";
 
+  /**
+   * Enables eTag checks for S3 PUT and MULTIPART upload requests.
+   */
+  public static final String S3_CHECKSUM_ENABLED = "s3.checksum.enabled";
```

Comment:
I know the conventions for property naming are a little mixed, but I believe this should be `s3.checksum-enabled`